lib/common/acl.c

```diff
@@ -262,14 +241,27 @@ pcmk__apply_acl(xmlNode *xml)
         max = numXpathResults(xpathObj);
 
         for (lpc = 0; lpc < max; lpc++) {
+            static struct qb_log_callsite *trace_cs = NULL;
             xmlNode *match = getXpathResult(xpathObj, lpc);
-            char *path = xml_get_path(match);
 
             p = match->_private;
-            crm_trace("Applying %s ACL to %s matched by %s",
-                      acl_to_text(acl->mode), path, acl->xpath);
             pcmk__set_xml_flags(p, acl->mode);
-            free(path);
+
+            /* Build a GString only if tracing is enabled.
+             * Can't use pcmk__log_else() because the else_action would be
+             * continue.
```

Comment:
We might want to create a `pcmk__log_if()` or `pcmk__log_then()` macro that acts like a flipped `pcmk__log_else()`: run this block only if we WOULD log a message. Would be a separate PR at this point if we did